lib/matplotlib/axes/_base.py

```diff
@@ -1275,7 +1275,11 @@ def __clear(self):
         for axis in self._axis_map.values():
             axis.clear()  # Also resets the scale to linear.
         for spine in self.spines.values():
+            # avoid expensive methods in the clearing of the spine, gh-26243
+            tmp_axis = spine.axis
+            spine.axis = None
```

Comment:
We have a context manager that will do this setting/resetting in `matplotlib.cbook._setattr_cm`.